scripts/west_commands/cmake.py

```diff
@@ -49,8 +49,10 @@ def run_cmake(args, cwd=None, capture_output=False):
             return None
     else:
         # A real error occurred, raise an exception
-        raise subprocess.CalledProcessError(cmd=p.args,
+        cpe = subprocess.CalledProcessError(cmd=p.args,
```

Comment:
With a detailed explanation why on slack, a better fix would be to pass the cmd and returncode arguments as positionals, which is what the constructor expects:

```
raise subprocess.CalledProcessError(p.args, p.returncode)
```

I assume this was done so that `run_build()` could support the new `capture_output` kwarg, which is also not present in west's cmake.py, but looking at the callers, nobody calls run_build() with `capture_output` that I can see, so why make this change at all compared to west's cmake module? Did I miss something?

In general:

It's my fault that during review I missed that this was not a straightforward copy of the west cmake module. However, when copying and modifying code somewhere else, it's always kinder on the reviewers to copy the code first, then make modifications in another commit, to make it easier for them to see what happened :).

Reply:
> raise subprocess.CalledProcessError(p.args, p.returncode)

I assume you mean

 raise subprocess.CalledProcessError(p.returncode, p.args)

> why make this change at all compared to west's cmake module? Did I miss something?

because `check_call` raises the exception, but `communicate()` does not.

>The bug is here, not in west's main. 

I disagree, I think west's main should not use `args` and instead reference `cpe.returncode` and `cpe.cmd`

Reply:
>> The bug is here, not in west's main.

> I disagree, I think both are at fault. This here was a bug, sure, but I think west's main should not use args and instead reference cpe.returncode and cpe.cmd

You're right, both are at fault.

>> why make this change at all compared to west's cmake module? Did I miss something?

> because check_call raises the exception, but communicate() does not.

I mean, why was `check_call` replaced with `communicate` when this code was moved over in the first place? The `capture_output` kwarg is not used as far as I can see -- that's what I'm asking if I missed. Is there any place this new kwarg is used that I did not notice?

Reply:
>  Is there any place this new kwarg is used that I did not notice?

I see this is used from `west boards` in your other comment, thanks.